common/scala/src/main/scala/whisk/core/entity/Identity.scala

```diff
@@ -38,13 +37,17 @@ object UserLimits extends DefaultJsonProtocol {
   implicit val serdes = jsonFormat3(UserLimits.apply)
 }
 
+protected[core] case class Namespace(name: EntityName, uuid: UUID)
+
+protected[core] object Namespace extends DefaultJsonProtocol {
+  implicit val serdes = jsonFormat2(Namespace.apply)
+}
+
 protected[core] case class Identity(subject: Subject,
-                                    namespace: EntityName,
+                                    namespace: Namespace,
```

Comment:
By having two uuids you allow for them to be different. Is that intended?

Reply:
The PR has to two motivations.
The first one is to make the (mis-)usage of the uuid as unique namespace alias explicit. 
The change  makes unrelated parts of the system (like for example authentication and user metrics)  also unrelated on the code level.  This could of course be done by providing new access functions to the Identity object. 
My second - not fully outspoken -  motivation for the change is to pave the path
to make the authentication extensible and/or exchangeable. For both cases it is imaginable 
the another authentication implementation  might not depend (fully) on the subject db but (additional) on an external identity and access management system that will not provide a uuid.

I plan to discuss a proposal for extending authentication and authorization on the dev list in the very near future. Since I saw this change more as driven by hygienic reasons (see my first motivation) I put it out separately.